### PR TITLE
Have admins confirm/reject potential matches

### PR DESCRIPTION
**Pull request by maurizi**
## Overview

Moves confirm/rejection of potential facility matches for most lists from being done by contributors to being done by OS Hub admins. Select contributors and/or lists can be changed in the Django admin to be moderated by contributors instead, as will any existing lists.

Also updates the admin list dashboard to add a new option to display all contributors (shown by default), as well as a new control for filtering by who is responsible for match moderation and pagination controls.

Connects #1979

## Demo

image

image

image

## Notes

This adds a new default `status` filter of `MATCHED` to the Dashboard - we'll restore the ability to see lists in other status' in https://github.com/open-apparel-registry/open-apparel-registry/issues/2017

## Testing Instructions

* On `develop`
* As a regular contributor, add a new list that will cause some items to be potential matches that need to be confirmed / rejected (I uploaded the same list twice w/ different names)
* On this branch run `scripts/manage migrate`
* You should be able to continue to moderate the lists you already uploaded
* If you upload another list, you should instead see a message that it is awaiting moderation
* Log out and log in as a super user (or log in using a private window). On the Dashboard lists page you should see the list you just uploaded, and be able to confirm/reject the potential match.
 * If you switch the match responsibility filter you should see the first w/ potential matches list you uploaded at the top of the list

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] This PR is targeted at the correct branch (`develop` vs. `ogr/develop`)
- [ ] If this PR applies to both OAR and OGR a companion PR has been created

### CODE REVIEW

File: src/app/src/components/FacilityListItems.jsx [hunk 1]
```diff
@@ -146,6 +149,13 @@ class FacilityListItems extends Component {
                 </p>
             ) : null;
 
+        const awaitingModerationMessage = readOnly ? (
+            <p style={{ color: 'red' }}>
+                This list has matches awaiting moderation from the{' '}
+                {isAdminUser ? 'contributor' : 'OS Hub admins'}.
```
Comment (maurizi) [inline src/app/src/components/FacilityListItems.jsx:155]: @obrienad this is the new copy we wanted to run by OAR - I took a stab at it (you see this in the 3rd screenshot on the PR description as well).

File: src/app/src/components/FacilityListItemsConfirmationTableRow.jsx [hunk 2]
```diff
@@ -110,7 +110,9 @@ function FacilityListItemsConfirmationTableRow({
                     variant="head"
                     style={listTableCellStyles.headerCellStyles}
                 >
-                    <b>Actions</b>
+                    <ShowOnly when={!readOnly}>
+                        <b>Actions</b>
+                    </ShowOnly>
```
Comment (maurizi) [inline src/app/src/components/FacilityListItemsConfirmationTableRow.jsx:115]: This is a change from how we displayed the read-only page before for admins, but it seemed a little weird to me to have an empty "Actions" header when we hide the actions themselves

**Comment by maurizi** [comment]
@jwalgran I assigned you as the primary reviewer because I think you have the most context on what is expected here, but @TaiWilkin was thinking of doing a follow-on task for this so I thought it might be helpful for them to just take a look at the code changes.

**Comment by jwalgran** [comment]
Great plan on the double review assignment. Thanks.

File: src/app/src/components/FacilityListItems.jsx [hunk 3]
```diff
@@ -162,7 +172,7 @@ class FacilityListItems extends Component {
             </Button>
         );
 
-        const backRoute = readOnly ? dashboardListsRoute : listsRoute;
+        const backRoute = isAdminUser ? dashboardListsRoute : listsRoute;
```
Comment (jwalgran) [inline src/app/src/components/FacilityListItems.jsx:175]: When an admin is returning to the list view using the in-app "Back to lists" button the selected contributor is being preserved but the new responsibility selection is not

**Review by jwalgran — APPROVED** [review]
Clear implementation. I did not find any issue in the core functionality and only a minor issue with admins losing filter selections when using the back button.

**Review by TaiWilkin — APPROVED** [review]
This looks good to me. Thanks for tagging me in to take a look. 
I was able to replicate the list navigation issue Justin found, but found no other issues.

**Pull request merged**